srml/system/src/lib.rs

```diff
@@ -404,8 +404,14 @@ impl<T: Trait> Module<T> {
 	/// Deposits an event into this block's event record adding this event
 	/// to the corresponding topic indexes.
 	///
-	/// This will update storage entries that correpond to the specified topics.
+	/// This will update storage entries that correspond to the specified topics.
 	/// It is expected that light-clients could subscribe to this topics.
+	///
+	/// # <weight>
+	/// - The `topics` argument affects complexity. 
+	/// - Each passed topic will be written to storage.
+	/// - Limited to `u32::max_value` topics.
```

Comment:
This is quite a large number, and it is iterated twice. Is `topics` typically limited at the origin of this call?

Reply:
Should be reviewed by someone who has better familiarity with the code. Adding a FLAG.

Reply:
it is not limited. however, this function is not publicly dispatchable so limiting it is left up to the caller.